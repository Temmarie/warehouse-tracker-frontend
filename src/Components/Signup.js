import React, { useState } from 'react';
import { createUser } from '../Actions';
import { useDispatch } from 'react-redux';

function Signup() {
  const dispatch = useDispatch();
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState(0);

  const handleChange = (e) => {
    let target = e.target.name
    let val = e.target.value
    if (target === 'username') {
      setUsername(val)
    }
    if (target === 'password') {
      setPassword(val)
    }
  }
  const handleSubmit = (e) => {
    e.preventDefault()
    dispatch(createUser(username, password))
  }

  return (
    <div>
      <h1>Signup</h1>
      <label
      >Username:<input onChange={e => handleChange(e)} type='name' name='username' required />
      </label>
      <label
      >Password:<input onChange={e => handleChange(e)} type='password' name='password' required />
      </label>
      <button onClick={e => handleSubmit(e)}>Sign Up</button>
    </div>
  );
}

export default Signup;