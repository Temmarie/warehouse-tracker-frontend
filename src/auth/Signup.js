/* eslint-disable */
import axios from "axios";
import React, { useState } from "react";
import { Redirect, useHistory } from "react-router";

const Signup = ({ user, setUser }) => {
  const [name, setName] = useState("");
  const [password, setPassword] = useState("");
  const history = useHistory();
  if (user === "Logged") {
    return <Redirect to={"/"} />;
  }

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const data = await axios.post(
        "https://warehouse-api-backend.herokuapp.com/api/v1/register",
        {
          user: {
            username,
            password,
            password_confirmation
          }
        }
      );
      localStorage.setItem("token", JSON.stringify(data.headers.authorization));
      data.data.message === "Signed up sucessfully."
        ? setUser("Logged")
        : setUser("Not Logged");
      history.push("/");
    } catch (error) {
      console.log(error);
    }
  };
  return (
    <div>
      <form
        onSubmit={handleSubmit}
      >
        <label htmlFor="email">
          name
        </label>
        <input
          onChange={(e) => setName(e.target.value)}
          name="name"
          type="text"
          placeholder="name"
          required
        />
        <label htmlFor="password">
          Password
        </label>
        <input
          onChange={(e) => setPassword(e.target.value)}
          name="password"
          type="password"
          placeholder="password"
          required
        />
        <button type="submit">
          Sign up
        </button>
        <a href="/login">
          Log in
        </a>
      </form>
    </div>
  );
};

export default Signup;