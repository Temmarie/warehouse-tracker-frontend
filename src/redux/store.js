import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import registrationReducer from './user/registrationReducer';

const store = createStore(registrationReducer, applyMiddleware(thunk, logger));

export default store;
