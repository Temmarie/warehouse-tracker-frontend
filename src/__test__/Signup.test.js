import { Provider } from 'react-redux';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Signup from '../Components/Signup';
import store from '../Reducers/index';

describe('<Signup />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Router><Provider store={store}><Signup /></Provider></Router>);
  });

  it('renders three <Nav /> components', () => {
    expect(wrapper.find('Nav')).toBeTruthy();
  });

  it('matches the snapshot', () => {
    const tree = renderer.create(
      <Router><Provider store={store}><Signup /></Provider></Router>,
    );
    expect(tree).toMatchSnapshot();
  });
});
