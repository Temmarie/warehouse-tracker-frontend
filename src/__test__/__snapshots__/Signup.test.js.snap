// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> matches the snapshot 1`] = `
<div>
  <nav
    aria-label="main navigation"
    className="navbar is-dark"
    role="navigation"
  >
    <div
      className="navbar-brand"
    >
      <div
        className="navbar-item"
      >
        <div
          className="navbar-item"
        >
          <svg
            fill="currentColor"
            height="40px"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "#4c79f2",
              }
            }
            viewBox="0 0 576 512"
            width="40px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M560 288h-80v96l-32-21.3-32 21.3v-96h-80c-8.8 0-16 7.2-16 16v192c0 8.8 7.2 16 16 16h224c8.8 0 16-7.2 16-16V304c0-8.8-7.2-16-16-16zm-384-64h224c8.8 0 16-7.2 16-16V16c0-8.8-7.2-16-16-16h-80v96l-32-21.3L256 96V0h-80c-8.8 0-16 7.2-16 16v192c0 8.8 7.2 16 16 16zm64 64h-80v96l-32-21.3L96 384v-96H16c-8.8 0-16 7.2-16 16v192c0 8.8 7.2 16 16 16h224c8.8 0 16-7.2 16-16V304c0-8.8-7.2-16-16-16z"
            />
          </svg>
        </div>
        <div
          className="navbar-item brand-name is-white"
        >
          WAREHOUSE
        </div>
      </div>
      <div
        className="navbar-burger navbar-end"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
          />
        </svg>
      </div>
      <div
        className="nav-menu"
      >
        <div
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          <a
            href="/Login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
        <div
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          <a
            href="/Signup"
            onClick={[Function]}
          >
            Signup
          </a>
        </div>
      </div>
    </div>
    <div
      className="navbar-menu"
    >
      <div
        className="navbar-end"
      >
        <div
          className="navbar-item"
        >
          <a
            href="/Login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
        <div
          className="navbar-item"
        >
          <a
            href="/Signup"
            onClick={[Function]}
          >
            Signup
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div
    className="login-form-container"
  >
    <div
      className="card width-400"
    >
      <div
        className="card-image"
      >
        <figure
          className="image"
        >
          <img
            alt="placeholder"
            src="background.jpg"
          />
        </figure>
      </div>
      <div
        className="card-content"
      >
        <div
          className="media"
        >
          <div
            className="is-size-4"
          >
            SignUp
          </div>
        </div>
        <div
          className="content"
        >
          <div
            className="field"
          >
            <label
              htmlFor="name"
            >
              <div
                className="control has-icons-left has-icons-right"
              >
                <input
                  className="input"
                  name="username"
                  onChange={[Function]}
                  placeholder="username"
                  required={true}
                  type="text"
                />
                <span
                  className="icon is-small is-left"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 448 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                    />
                  </svg>
                </span>
              </div>
            </label>
          </div>
          <div
            className="field"
          >
            <label
              htmlFor="password"
            >
              <div
                className="control has-icons-left has-icons-right"
              >
                <input
                  className="input"
                  name="password"
                  onChange={[Function]}
                  placeholder="password"
                  required={true}
                  type="password"
                />
                <span
                  className="icon is-small is-left"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12,2C9.243,2,7,4.243,7,7v2H6c-1.103,0-2,0.897-2,2v9c0,1.103,0.897,2,2,2h12c1.103,0,2-0.897,2-2v-9c0-1.103-0.897-2-2-2 h-1V7C17,4.243,14.757,2,12,2z M9,7c0-1.654,1.346-3,3-3s3,1.346,3,3v2H9V7z M18.002,20H13v-2.278c0.595-0.347,1-0.985,1-1.722 c0-1.103-0.897-2-2-2s-2,0.897-2,2c0,0.736,0.405,1.375,1,1.722V20H6v-9h12L18.002,20z"
                    />
                  </svg>
                </span>
              </div>
            </label>
            <div
              className="field padding-2-5"
            >
              <button
                className="button is-fullwidth is-success"
                onClick={[Function]}
                type="button"
              >
                Sign Up
              </button>
            </div>
          </div>
          <time
            dateTime="2016-1-1"
          >
            11:09 PM - 1 Jan 2016
          </time>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Signup /> matches the snapshot 1`] = `
<div>
  <nav
    aria-label="main navigation"
    className="navbar is-dark"
    role="navigation"
  >
    <div
      className="navbar-brand"
    >
      <div
        className="navbar-item"
      >
        <div
          className="navbar-item"
        >
          <svg
            fill="currentColor"
            height="40px"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "#4c79f2",
              }
            }
            viewBox="0 0 576 512"
            width="40px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M560 288h-80v96l-32-21.3-32 21.3v-96h-80c-8.8 0-16 7.2-16 16v192c0 8.8 7.2 16 16 16h224c8.8 0 16-7.2 16-16V304c0-8.8-7.2-16-16-16zm-384-64h224c8.8 0 16-7.2 16-16V16c0-8.8-7.2-16-16-16h-80v96l-32-21.3L256 96V0h-80c-8.8 0-16 7.2-16 16v192c0 8.8 7.2 16 16 16zm64 64h-80v96l-32-21.3L96 384v-96H16c-8.8 0-16 7.2-16 16v192c0 8.8 7.2 16 16 16h224c8.8 0 16-7.2 16-16V304c0-8.8-7.2-16-16-16z"
            />
          </svg>
        </div>
        <div
          className="navbar-item brand-name is-white"
        >
          WAREHOUSE
        </div>
      </div>
      <div
        className="navbar-burger navbar-end"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
          />
        </svg>
      </div>
      <div
        className="nav-menu"
      >
        <div
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          <a
            href="/Login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
        <div
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          <a
            href="/Signup"
            onClick={[Function]}
          >
            Signup
          </a>
        </div>
      </div>
    </div>
    <div
      className="navbar-menu"
    >
      <div
        className="navbar-end"
      >
        <div
          className="navbar-item"
        >
          <a
            href="/Login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
        <div
          className="navbar-item"
        >
          <a
            href="/Signup"
            onClick={[Function]}
          >
            Signup
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div
    className="login-form-container"
  >
    <div
      className="card width-400"
    >
      <div
        className="card-image"
      >
        <figure
          className="image"
        >
          <img
            alt="placeholder"
            src="background.jpg"
          />
        </figure>
      </div>
      <div
        className="card-content"
      >
        <div
          className="media"
        >
          <div
            className="is-size-4"
          >
            SignUp
          </div>
        </div>
        <div
          className="content"
        >
          <div
            className="field"
          >
            <label
              htmlFor="name"
            >
              <div
                className="control has-icons-left has-icons-right"
              >
                <input
                  className="input"
                  name="username"
                  onChange={[Function]}
                  placeholder="username"
                  required={true}
                  type="text"
                />
                <span
                  className="icon is-small is-left"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 448 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                    />
                  </svg>
                </span>
              </div>
            </label>
          </div>
          <div
            className="field"
          >
            <label
              htmlFor="password"
            >
              <div
                className="control has-icons-left has-icons-right"
              >
                <input
                  className="input"
                  name="password"
                  onChange={[Function]}
                  placeholder="password"
                  required={true}
                  type="password"
                />
                <span
                  className="icon is-small is-left"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12,2C9.243,2,7,4.243,7,7v2H6c-1.103,0-2,0.897-2,2v9c0,1.103,0.897,2,2,2h12c1.103,0,2-0.897,2-2v-9c0-1.103-0.897-2-2-2 h-1V7C17,4.243,14.757,2,12,2z M9,7c0-1.654,1.346-3,3-3s3,1.346,3,3v2H9V7z M18.002,20H13v-2.278c0.595-0.347,1-0.985,1-1.722 c0-1.103-0.897-2-2-2s-2,0.897-2,2c0,0.736,0.405,1.375,1,1.722V20H6v-9h12L18.002,20z"
                    />
                  </svg>
                </span>
              </div>
            </label>
            <div
              className="field padding-2-5"
            >
              <button
                className="button is-fullwidth is-success"
                onClick={[Function]}
                type="button"
              >
                Sign Up
              </button>
            </div>
          </div>
          <time
            dateTime="2016-1-1"
          >
            11:09 PM - 1 Jan 2016
          </time>
        </div>
      </div>
    </div>
  </div>
</div>
`;
